### PR TITLE
feat: allow user to redirect after signin

### PR DESCRIPTION
**Pull request by ojeytonwilliams**

### CODE REVIEW

File: api-server/server/component-passport.js [hunk 1]
```diff
@@ -81,9 +81,8 @@ export const saveResponseAuthCookies = () => {
 export const loginRedirect = () => {
   return (req, res) => {
     const successRedirect = req => {
-      if (!!req && req.session && req.session.returnTo) {
-        delete req.session.returnTo;
-        return `${homeLocation}/learn`;
+      if (req && req.query && req.query.returnTo) {
+        return req.query.returnTo;
```
Comment (raisedadead) [inline api-server/server/component-passport.js:85]:
```suggestion
      if (req && req.session && req.session.returnTo) {
      	delete req.session.returnTo;
      }

      if (req && req.query && req.query.returnTo) {
        return req.query.returnTo;
```

Just a failsafe if the session key ever gets set. We might as well clean it up? What do you think?

**Review by raisedadead — CHANGES_REQUESTED** [review]

File: api-server/server/component-passport.js (hunk 1 above)
Comment (ojeytonwilliams) [reply api-server/server/component-passport.js:85]: We do set the session key
https://github.com/freeCodeCamp/freeCodeCamp/blob/master/api-server/server/middlewares/add-return-to.js

but we don't use it.

There's no harm in deleting it, so sure, but we should remove or use that middleware, too.

Comment (raisedadead) [reply api-server/server/component-passport.js:89]: Right that whole thing should go.

Comment (ojeytonwilliams) [reply api-server/server/component-passport.js:85]: I checked there was nothing worth salvaging from it (there wasn't) and took it out.

**Pull request merged**